### ISSUE TITLE
Allowing `each` and `into` on a vec

### ISSUE THREAD

**Issue by Freyskeyd** [colin-kiegel/rust-derive-builder#209]
Hello,

I'm trying to have a nice api for one of my crates and I was wondering if I could be able to have a struct like this:

```rust

#[derive(derive_builder::Builder, Debug, Clone)]
#[builder(setter(into))]
struct SessionManagementPacketResult {
    #[builder(default = "SessionState::Opening")]
    session_state: SessionState,
    #[builder(setter(each = "packet", into))]
    packets: Vec<Packet>,
}
```

I want to be able to do `session.packet(xx).packet(xx)` where `xx` implement `From<xx> for Packet` (the into part) but it seems to not work, or the `into` isn't applied to `packet`.

Any idea of a workaround on that? 

Thank's!

**Comment by TedDriggs** [comment]
I believe the issue there is that `Extend` has its own type coercion magic going on and stacking the two breaks inference. See [this discussion](https://github.com/colin-kiegel/rust-derive-builder/pull/199#discussion_r596022844) from the initial PR to add `each`.

**Comment by TedDriggs** [comment]
This is fixed by #234
